scanner_cli.py

```diff
@@ -63,40 +69,93 @@ def fetch_properties(state_code):
+
+        # Extract specific information from each property
+        filtered_properties = [
+            {
+                "property_id": property["property_id"],
+                "address": {
+                    "line": property["location"]["address"]["line"],
+                    "city": property["location"]["address"]["city"],
+                    "state_code": property["location"]["address"]["state_code"],
+                    "postal_code": property["location"]["address"]["postal_code"]
+                },
+                "list_price": property["list_price"],
+                "description": {
+                    "beds": property["description"]["beds"],
+                    "baths": property["description"]["baths"]
+                },
+                "primary_photo": property["primary_photo"]
+            }
+            for property in property_list
+        ]
+
+        # Write to JSON
+        with open(file_name, mode='w', encoding="utf-8") as file:
+            json.dump(filtered_properties, file, indent=2)
 
         print(f"Retrieved newest 42 properties for sale in {state_code}")
         print(f"Saved to {file_name}")
     except requests.exceptions.RequestException as ex:
         print(f"Error: {ex}")
 
+def store_user_data(username):
+    """
+    Store and calculate metrics based on user data.
+
+    Parameters:
+    - username (str): Username of the user, used as a reference for stored values.
+
+    Returns:
+    None: Updates the json file user_data.json
+    """
+    # Load existing data from file
+    try:
+        with open('user_data.json', 'r', encoding="utf8") as file:
+            user_data = json.load(file)
+        # Check if the username is already in the database
+        if username in user_data:
+            print(f"Hello, {username}!")
+        else:
+            print(f"New user {username}")
+    except FileNotFoundError:
+        # Create a new user_data dictionary if the file doesn't exist
+        user_data = {}
+        print(f"New user {username}")
+
+    # Ask for and save user input
+    user_data[username] = {
+        'min_price_range': float(input("Enter your minimum price $: ")),
+        'max_price_range': float(input("Enter your maximum price $: ")),
+        'school_rating_preference': int(input("Enter your school rating preference (1-5): ")),
+        'gross_income': float(input("Enter your gross income $: ")),
+        'current_debt': float(input("Enter your current debt $: ")),
+        'property_management_fees': float(input("Enter property management fees as percent of income: ")),
+        'maintenance_costs': float(input("Enter maintenance costs as percent of income: ")),
+        'insurance': float(input("Enter insurance costs: "))
+    }
+
+    # Save the updated user data
+    with open('user_data.json', 'w', encoding="utf8") as file:
+        json.dump(user_data, file, indent=2)
+
+    print(f"User data for {username} has been updated.")
+
 if __name__ == "__main__":
```

Comment:
nit: suggest using [Click](https://click.palletsprojects.com/en/8.1.x/) for cmd line interface

Reply:
Using the module argparse to handle CLI commands, but I'm printing out info for more verbose output